'''
2. Создайте собственный класс-исключение, обрабатывающий ситуацию деления на нуль.
Проверьте его работу на данных, вводимых пользователем. При вводе пользователем нуля в качестве делителя программа
должна корректно обработать эту ситуацию и не завершиться с ошибкой.
'''
class MyException(Exception):
    def __init__(self, txt):
        self.txt = txt
        return None

a = int(input('Введите первое число: '))
b = int(input('Введите второе число: '))

try:
    if b == 0:
        raise MyException('Ошибка деления на ноль. В принципе, делить на ноль нельзя, но когда очень хочется, то можно...')
    c = a / b
except MyException as err:
    print(f'{err}')
    c = 0
finally:
    print(f'Результат деления равен: {c}')


